src/viperleed/calc/files/parameters/file_reader.py

```diff
@@ -0,0 +1,61 @@
+"""Module file_reader of viperleed.calc.classes.
+
+This module is a more abstract version of a settings file reader. It is based on
+@michele-riva's work on the PARAMETERS file reader, which in turn is based on
+the old readPARAMETERS from @fkraushofer.
+
+Defines context-manager, iterator classes for reading parameters from a file.
+"""
+
+__authors__ = (
+    'Alexander M. Imre (@amimre)',
+    'Michele Riva (@michele-riva)',
+    )
+__copyright__ = 'Copyright (c) 2019-2024 ViPErLEED developers'
+__created__ = '2024-10-03'
+__license__ = 'GPLv3+'
+
+from abc import abstractmethod
+from collections.abc import Iterator
+from contextlib import AbstractContextManager
+from pathlib import Path
+
+
```

Comment:
I can see some useful subclasses for this one. An example that comes to mind is a `DelimitedInputFileReader` that takes care of files where lines are of the form:

`<stuff on the left> <delimiter> <stuff on the right>`

Another useful subclass that may require a bit of generalization of this one: a reader for files that are split into "blocks". Then each block could of course be a reader itself. In that case, generalization to a stream would be ideal.